src/Domain.LinnApps/PurchaseOrders/PurchaseOrderDeliveryService.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace Linn.Purchasing.Domain.LinnApps.PurchaseOrders
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Linn.Common.Persistence;
+    using Linn.Purchasing.Domain.LinnApps.Keys;
+
+    public class PurchaseOrderDeliveryService : IPurchaseOrderDeliveryService
+    {
+        private readonly IRepository<PurchaseOrderDelivery, PurchaseOrderDeliveryKey> repository;
+
+        public PurchaseOrderDeliveryService(IRepository<PurchaseOrderDelivery, PurchaseOrderDeliveryKey> repository)
+        {
+            this.repository = repository;
+        }
+
+        public IEnumerable<PurchaseOrderDelivery> SearchDeliveries(
+            string supplierSearchTerm, string orderNumberSearchTerm, bool includeAcknowledged)
+        {
+            var result = this.repository.FindAll();
```

Comment:
findAll initially seemed very bold but because you're not .toList ing etc is it just when it gets to the order by it'll evaluate, and by that time have some wheres on it to cut it down?

Reply:
yeah.. its just so I can build up the query in the domain service and test it since there's a bit of logic involved which would otherwise just get hidden away in the FilterBy(expression)